src/main/java/xyz/aadev/generitech/common/tileentities/power/TileEntityPower.java

```diff
@@ -26,32 +27,51 @@
 import xyz.aadev.aalib.common.inventory.InventoryOperation;
 import xyz.aadev.aalib.common.tileentities.TileEntityInventoryBase;
 import xyz.aadev.generitech.Reference;
+import xyz.aadev.generitech.client.gui.machines.GuiPulverizer;
+import xyz.aadev.generitech.client.gui.power.GuiGenerator;
+import xyz.aadev.generitech.common.container.machines.ContainerPulverizer;
+import xyz.aadev.generitech.common.container.power.ContanierGenerator;
+import xyz.aadev.generitech.common.tileentities.TileEntityMachineBase;
 
 import javax.annotation.Nullable;
 import java.util.List;
 
-public class TileEntityPower extends TileEntityInventoryBase implements ITeslaProducer, net.minecraft.util.ITickable, IWailaBodyMessage {
+public class TileEntityPower extends TileEntityMachineBase implements ITeslaProducer, net.minecraft.util.ITickable, IWailaBodyMessage {
     private BaseTeslaContainer container = new BaseTeslaContainer(0, 50000, 1000, 1000);
     private InternalInventory inventory = new InternalInventory(this, 1);
-    private int T0transfer = 100;
+    private int T0transfer = 120;
     private int fuelRemaining = 0;
     private Item lastFuelType;
     private int lastFuelValue;
+    private int fuelTotal = 0;
+
+    @Override
+    public Object getClientGuiElement(int guiId, EntityPlayer player) {
+        return new GuiGenerator(player.inventory, this);
+    }
+
+    @Override
+    public Object getServerGuiElement(int guiId, EntityPlayer player) {
+        return new ContanierGenerator(player.inventory, this);
+    }
 
     @Override
     public long takePower(long power, boolean simulated) {
         return power;
     }
 
 
+    public long powerStored(){
+        return container.getStoredPower();
+    }
 
     @Override
     public void update() {
         BlockPos pos = getPos();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "pos" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "pos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1854)